docs/design/secrets.md

```diff
@@ -0,0 +1,486 @@
+# Secret Distribution
+
+## Abstract
+
+A proposal for the distribution of secrets (passwords, keys, etc) to the Kubelet and to
+containers inside Kubernetes using a custom volume type.
+
+## Motivation
+
+Secrets are needed in containers to access internal resources like the Kubernetes master or
+external resources such as git repositories, databases, etc. 
+
+A goal of this design is to eliminate or minimize the modifications to containers in order to
+access secrets. Secrets should be placed where the container expects them to be.
+
+## Constraints and Assumptions
+
+*  This design does not prescribe a method for storing secrets; storage of secrets should be
+   pluggable to accomodate different use-cases
+*  Encryption of secret data and node security are orthogonal concerns
+*  It is assumed that node and master are secure and that compromising their security could also
+   compromise secrets:
+   *  If a node is compromised, only the secrets for the containers scheduled on it should be
```

Comment:
This almost sounds like it's saying that if you compromise a node, you should be granted access to its containers' secrets. I know this is trying to say that a node must not hold on to secrets for containers that don't live on it. Maybe something like this would work a bit better?

If a node is compromised, the only secrets that could potentially be exposed should (must?) belong to containers scheduled on it.